libr/bin/p/bin_dex.c

```diff
@@ -859,21 +844,21 @@ static char *dex_field_name(RBinDexObj *bin, int fid) {
 
 static char *dex_method_fullname(RBinDexObj *bin, int method_idx) {
 	if (!bin || !bin->types) {
-		return NULL;
+		return strdup ("");
 	}
 	if (method_idx < 0 || method_idx >= bin->header.method_size) {
-		return NULL;
+		return strdup ("");
 	}
 	int cid = bin->methods[method_idx].class_id;
 	if (cid < 0 || cid >= bin->header.types_size) {
-		return NULL;
+		return strdup ("");
 	}
 	char *name = dex_method_name (bin, method_idx);
-	char *class_name = dex_class_name_byid (bin, cid);
+	char *class_name = strdup (dex_class_name_byid (bin, cid));
```

Comment:
Without the strdup it crashes, because "r_str_replace" does a free on all params.